compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3059,6 +3059,10 @@ private BRecordType createErrorDetailRecordType(BLangInvocation iExpr, boolean r
     }
 
     private void checkObjectFunctionInvocationExpr(BLangInvocation iExpr, BObjectType objectType) {
+        if (Names.SERVICE.value.equals(objectType.tsymbol.name.value)) {
```

Comment:
Shall we check flag here.? instead of the name.

Reply:
The `objectType` is of type built in service type and does not have any flags or tags.

Reply:
`objectType` is of builtin type service, there is no flags/tags on it, but I have refactored to use the `getKind()` function instead of name.